PR: Icon Component Article

bbrennanbasis opened the pull request:
This PR closes #226. 

Changes in this PR included:
- Created icon.md
- Deleted dwc-icon.md
- Added a hyperlink to icon.md on the component overview page

**Note**: There will be merge conflicts with this and PR #280 because of the deletion of dwc-icon.md.

hyyan commented on docs/components/icon.md:
```diff
@@ -0,0 +1,194 @@
+An icon pool is a collection of commonly used icons that enables easy access and reuse. By using icons from an icon pool, you can ensure that the icons in your app are recognizable and share a consistent style.
+Using webforJ allows you to choose from three pools, or implement a custom pool.
+Each pool has an extensive collection of open source icons that are free to use.
+WebforJ gives you the flexibility to choose from three pools and use them as unique classes, without the hassle of downloading any of the icons directly.
```
should be `webforJ`

hyyan commented on docs/components/icon.md:
````diff
@@ -0,0 +1,194 @@
+
+```java
+// Create an icon from a String name
+Icon image = FeatherIcon.create("image");
````
`FeatherIcon` builder does not allow to create an icon from string name.

hyyan commented on docs/components/icon.md:
````diff
@@ -0,0 +1,194 @@
+Icon music = FontAwesomeIcon.create("user", FontAwesomeIcon.Variate.SOLID);
+```
+
+## Adding icons to components
````
This section should be rephrased to avoid listing the specific components that support prefix and suffix slots, as more components will support this in the future. Instead, it should be more general, stating that some components currently support this feature that implement the `HasPrefixAndSuffix` interface, allowing for the use of prefix and suffix icons or other valid components.

hyyan commented on docs/components/icon.md:
```diff
@@ -0,0 +1,194 @@
+
+Ultimately, consistency is key. Once you choose a style, maintain it across your site for a cohesive and user-friendly design.
+
+## Creating custom pools
```
This will need working code sample. Check https://github.com/webforj/webforj/pull/717

hyyan replied:
@bbrennanbasis, this should be a code sample only, not a complete runnable demo.

hyyan commented on docs/components/icon.md:
````diff
@@ -0,0 +1,194 @@
+An `Icon` component is nonselectable, but for actions that are best represented with just an icon, such as notifications or alerts, you can use the `IconButton`.
+
+ ```java
+ FlexLayout layout = FlexLayout.create().horizontal().build().setMargin("var(--dwc-space-m)");
````
show only the relevant part:

```java
IconButton bell = new IconButton(FeatherIcon.BELL.create());
bell.onClick(e -> showMessageDialog("You have a new message!", "Ding Dong!"));
```

hyyan commented on docs/components/icon.md:
```diff
@@ -0,0 +1,194 @@
+- `GRAY`: Good for subtle actions, such as minor settings or actions that are more supplementary to a page, and not part of the main functionality.
+- `INFO`: Good for providing additional clarifying information to a user.
+
+### Expanses
```
should the `Below is information on the attributes, as well as parts and properties required for customizing icons with CSS.` be removed , why it is in here ?

bbrennanbasis replied:
The Expanses section was text from [dwc-icon styling](https://documentation.webforj.com/docs/components/dwc-icon#expanses). I'll go ahead and remove it.

hyyan commented:
@bbrennanbasis Overall, everything looks good. I've requested a few changes. Also, please address all the Vale issues.  
@MatthewHawkins The article still needs a few runnable demos. Who will be working on those?

MatthewHawkins commented:
@laurenic0l Would you please read through this article, determine one or two spots for demo programs to be created, and then create them please?

@bbrennanbasis Let me know if you want a hand with any of the Vale comments - the components imports should be able to be removed as they're now globally available to all articles, and I'm happy to help with any others.

bbrennanbasis commented:
@hyyan Included your review request, rebased, and included a link on the `Button` article.

This PR now also closes #177 and #287.

hyyan commented:
@laurenic0l, please include the demo URLs once you're finished.

hyyan commented:
@bbrennanbasis could merge these two sections and streamline the wording a bit: "Icons before component text" and "Icons after component text."

@laurenic0l The sections "Icons before component text" and "Icons after component text" need an sample.

bbrennanbasis commented:
@hyyan I've added the custom pool code snippet and consolidated the "Icons before component text" and "Icons after component text" sections to "Icon placement: before VS after".

hyyan commented on docs/components/icon.md:
```diff
@@ -0,0 +1,198 @@
+
+Ultimately, consistency is key. Once you choose a style, maintain it across your site for a cohesive and user-friendly design.
+
+   **Examples before text**
+   - 💾 Save
+   - ✏️ Edit
+   - 🗑️ Delete
+
+   **Examples after text**
+   - Next ➡️
+   - Options ☰
+   - Download ⬇
```
Remove these examples please , we a demo for this now

bbrennanbasis commented:
> @bbrennanbasis One last thing I noticed that’s still missing—could you briefly cover the Custom Pool Factory as well?

@hyyan I've added a section for this and removed the before and after examples using emojis.

MatthewHawkins reviewed (dismissed):
Custom Icon pool section looks clear to me
